core/domain/image_services_test.py

```diff
@@ -47,13 +48,15 @@ def setUp(self):
             mode='rb', encoding=None) as f:
             self.png_raw_image = f.read()
 
-    def test_image_dimensions_outputs_correctly(self):
+    def test_image_dimensions_are_output_correctly(self):
         height, width = (
             image_services.get_image_dimensions(self.jpeg_raw_image))
         self.assertEqual(self.TEST_IMAGE_HEIGHT, height)
         self.assertEqual(self.TEST_IMAGE_WIDTH, width)
 
-    def test_dev_mode_returns_regular_image(self):
+    def test_dev_mode_returns_original_uncompressed_image(self):
+        # Test to test the behavior when we do not set constants.DEV_MODE
```

Comment:
I'm still a bit confused about what all this stuff has got to do with "dev mode" (and why it's related in the first place). Could you explain?

Reply:
Basically the function returns the original image if currently in DEV Mode here:
`    if not constants.DEV_MODE:
        if scaling_factor > 1 or scaling_factor <= 0:
            raise ValueError('Scaling factor should be in the interval (0, 1].')
        image = Image.open(io.BytesIO(image_content))

        image_format = image.format
        height, width = _get_pil_image_dimensions(image)
        new_width = int(width * scaling_factor)
        new_height = int(height * scaling_factor)
        new_image_dimensions = (new_width, new_height)

        # NOTE: image.thumbnail() function does not work when the scale factor
        # is greater than 1.
        image.thumbnail(new_image_dimensions, Image.ANTIALIAS)
        with io.BytesIO() as output:
            image.save(output, format=image_format)
            new_image_content = output.getvalue()
        return new_image_content
    else:
        return image_content`
This was part of the original API, this test is to make sure that if it is currently DEV_MODE, I will make sure that's clear by adding a swap.

Reply:
Don't know why the formatting is screwed up.

Reply:
Ah -- use three backticks in a row to denote multiline monospace font. ```

You can click on "Preview" to make sure it shows up correctly.

Reply:
Ah fixed!